src/colony/lua_tm.c

```diff
@@ -261,9 +261,14 @@ static int l_tm_tcp_read (lua_State* L)
   uint8_t buf[512];
   size_t buf_len = sizeof(buf);
   int err = tm_tcp_read(socket, buf, &buf_len);
-  (void) err;
+
+  if (err < 0) {
```

Comment:
I suspect this is wrong. `tm_tcp_read` returns `errno` on error:
https://github.com/tessel/runtime/blob/master/src/posix/tm_net.c#L169

…which for the POSIX-defined ones are mandated to be *positive* values: http://pubs.opengroup.org/onlinepubs/009695399/basedefs/errno.h.html

Basically any non-zero value indicates error.